internal/parse/parser.go

```diff
@@ -277,6 +303,137 @@ func (p *Parser) parseGoFullName() (FullName, bool, error) {
 	return FullName{}, false, nil
 }
 
+// parseList takes a parsing function that returns a FullName and parses a
+// bracketed, comma seperated, list. On success it returns an array of FullName
+// objects. Otherwise it returns an empty array, false and an error.
+func (p *Parser) parseList(parseFn func(p *Parser) (FullName, bool, error)) ([]FullName, bool, error) {
+	cp := p.save()
+	if p.skipByte('(') {
+		parenPos := p.pos
+		p.skipSpaces()
+
+		nextItem := true
+		var objs []FullName
+		for nextItem {
+			if obj, ok, err := parseFn(p); ok {
+				objs = append(objs, obj)
+			} else if err != nil {
+				return nil, false, err
+			} else {
+				return nil, false, fmt.Errorf("invalid identifier near char %d", p.pos)
+			}
+			p.skipSpaces()
+			if p.skipByte(')') {
+				return objs, true, nil
+			}
+			nextItem = p.skipByte(',')
+			p.skipSpaces()
+		}
+		return nil, false, fmt.Errorf("missing closing parentheses for char %d", parenPos)
+	}
+	cp.restore()
+	return nil, false, nil
+}
+
+// parseColumns parses text in the SQL query of the form "table.colname". If
+// there is more than one column then the columns must be enclosed in brackets
+// e.g.  "(col1, col2) AS &Person.*".
+func (p *Parser) parseColumns() ([]FullName, bool) {
+	cp := p.save()
+	// We skip a space here to keep consistent with parseTargets which also
+	// consumes one space before the start of the expression.
+	p.skipByte(' ')
+	// Case 1: A single column e.g. p.name
+	if col, ok, _ := p.parseColumn(); ok {
+		return []FullName{col}, true
+	} else if cols, ok, _ := p.parseList((*Parser).parseColumn); ok {
+		return cols, true
+	}
```

Comment:
This would be equivalent and a bit more readable maybe? We could easily add a comment for the list case.
```       
        // Case 1: A single column e.g. p.name
        if col, ok, _ := p.parseColumn(); ok {
                return []FullName{col}, true
        }

        // Case 2: Multiple columns
        if cols, ok, _ := p.parseList((*Parser).parseColumn); ok {
                return cols, true
        }
```

Reply:
Changed, and make the examples in the Case comments consistent.